sites/public/src/lib/listings/search.ts

```diff
@@ -0,0 +1,121 @@
+import { ListingQueryBuilder } from "./listing-query-builder"
+
+export type ListingSearchParams = {
+  bedrooms: string
+  bathrooms: string
+  monthlyRent: string
+  counties: string[]
+}
+
+/**
+ * Parses a search string into an object
+ *
+ * The format object describes the expected output.  If the value of a property
+ * in the template is an array, it will attempt to split the value on commas and
+ * assign the result. If not, it will assign the result directly as a string value.
+ *
+ * @param format
+ * @param search
+ * @returns
+ */
+export function parseSearchString<T extends object>(search: string, format: T): T {
```

Comment:
The `format` arg here defines what to expect within the search string.  If a prop isn't present, it doesn't get added even if it's in the search string.  If the prop in the `format` object has an array value, it combines everything into an array.

I was hoping to be able to use a type (ie `ListingSearchParams`) rather than an object for defining the shape of the output, but I didn't see a way to do this without reflection or something equally hacky.  Open to suggestions.